frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -141,59 +171,490 @@ describe("AddToDashSelectDashModal", () => {
     });
 
     describe("when user visited some dashboard in last 24hrs", () => {
       it("should preselected last visited dashboard in the picker", async () => {
-        await setup();
+        await setup({
+          mostRecentlyViewedDashboard: DASHBOARD,
+        });
 
-        const dashboardCollection = COLLECTIONS.find(
-          collection => collection.id === DASHBOARD.collection_id,
+        const dashboardCollection = checkNotNull(
+          COLLECTIONS.find(
+            collection => collection.id === DASHBOARD.collection_id,
+          ),
         );
 
         // breadcrumbs
-        expect(
-          screen.getByText(`${dashboardCollection?.name}`),
-        ).toBeInTheDocument();
-        // dashboard item
+        assertBreadcrumbs([dashboardCollection]);
         expect(screen.getByText(DASHBOARD.name)).toBeInTheDocument();
       });
 
       describe("when last visited dashboard belongs to root", () => {
         it("should render root collection", async () => {
           await setup({
             dashboard: DASHBOARD_AT_ROOT,
+            mostRecentlyViewedDashboard: DASHBOARD_AT_ROOT,
           });
 
           // breadcrumbs
-          expect(screen.getByText(ROOT_COLLECTION.name)).toBeInTheDocument();
-          // dashboard item
+          assertBreadcrumbs([ROOT_COLLECTION]);
           expect(screen.getByText(DASHBOARD_AT_ROOT.name)).toBeInTheDocument();
         });
       });
+
+      describe("when last visited dashboard belongs to public subcollections", () => {
+        const dashboardInPublicSubcollection = createMockDashboard({
+          id: 5,
+          name: "Dashboard in public subcollection",
+          collection: SUBCOLLECTION,
+          collection_id: SUBCOLLECTION.id as number,
+          model: "dashboard",
+        });
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
+          await setup({
+            mostRecentlyViewedDashboard: dashboardInPublicSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([ROOT_COLLECTION, COLLECTION, SUBCOLLECTION]);
+
+          // dashboard item
+          expect(
+            screen.getByText(dashboardInPublicSubcollection.name),
+          ).toBeInTheDocument();
+        });
+
+        it("should show the root collection if the question is in a personal collection", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            mostRecentlyViewedDashboard: dashboardInPublicSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([ROOT_COLLECTION]);
+
+          // Showing personal collection means we're viewing the root collection
+          expect(
+            screen.getByRole("heading", { name: PERSONAL_COLLECTION.name }),
+          ).toBeInTheDocument();
+        });
+      });
+
+      describe("when last visited dashboard belongs to personal subcollections", () => {
```

Comment:
test Query builder > Newly created question > #8, #10 from [the epic](https://github.com/metabase/metabase/issues/34733)